internal/services/solution_checker_rules/datasource_solution_checker_rules.go

```diff
@@ -0,0 +1,161 @@
+							Description:         "URL to detailed guidance on addressing the issue",
+							MarkdownDescription: "URL to detailed guidance on addressing the issue",
+							Computed:            true,
+						},
+						"component_type": schema.Int64Attribute{
+							Description:         "The type of component this rule applies to",
+							MarkdownDescription: "The type of component this rule applies to",
+							Computed:            true,
+						},
+						"primary_category": schema.Int64Attribute{
+							Description:         "The primary category of the rule",
+							MarkdownDescription: "The primary category of the rule",
+							Computed:            true,
+						},
+						"primary_category_description": schema.StringAttribute{
+							Description:         "Description of the primary category",
+							MarkdownDescription: "Description of the primary category",
+							Computed:            true,
+						},
+						"include": schema.BoolAttribute{
+							Description:         "Whether the rule is included/enabled by default",
+							MarkdownDescription: "Whether the rule is included/enabled by default",
+							Computed:            true,
+						},
+						"severity": schema.Int64Attribute{
+							Description:         "The severity level of the rule (1-5, with 5 being most severe)",
+							MarkdownDescription: "The severity level of the rule (1-5, with 5 being most severe)",
+							Computed:            true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *DataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	ctx, exitContext := helpers.EnterRequestContext(ctx, d.TypeInfo, req)
+	defer exitContext()
+	if req.ProviderData == nil {
+		// ProviderData will be null when Configure is called from ValidateConfig. It's ok.
+		return
+	}
+	client, ok := req.ProviderData.(*api.ProviderClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected ProviderData Type",
+			fmt.Sprintf("Expected *api.ProviderClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	d.SolutionCheckerRulesClient = newSolutionCheckerRulesClient(client.Api)
+}
+
+func (d *DataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	ctx, exitContext := helpers.EnterRequestContext(ctx, d.TypeInfo, req)
+	defer exitContext()
+
+	var state DataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	environmentId := state.EnvironmentId.ValueString()
+	rules, err := d.SolutionCheckerRulesClient.GetSolutionCheckerRules(ctx, environmentId)
+	if err != nil {
+		resp.Diagnostics.AddError(fmt.Sprintf("Client error when reading %s", d.ProviderTypeName), err.Error())
+		return
+	}
+
+	state.Rules = []RuleModel{}
+	for _, rule := range rules {
+		ruleModel := convertFromRuleDto(rule)
+		state.Rules = append(state.Rules, ruleModel)
+	}
+
```

Comment:
we should have a debug log here with info that we finished to logic

Reply:
exit context already does that.